controller/volume_controller.go

```diff
@@ -1145,6 +1150,118 @@ func (vc *VolumeController) ReconcileVolumeState(v *longhorn.Volume, es map[stri
 	return nil
 }
 
+func (vc *VolumeController) ReconcileLocalReplica(v *longhorn.Volume, e *longhorn.Engine, rs map[string]*longhorn.Replica) error {
+	keepLocalReplicaPolicy, err := vc.ds.GetSettingAsBool(types.SettingNameKeepLocalReplicaOnSameNodeAsEngine)
+	if err != nil {
+		return err
+	}
+	if !keepLocalReplicaPolicy {
+		return nil
+	}
+
+	if v.Status.State != types.VolumeStateAttached {
+		return nil
+	}
+
+	if e == nil {
+		return fmt.Errorf("BUG: handleKeepLocalReplicaWithEngine needs a valid engine")
```

Comment:
`return nil`
IIRC, for the newly created volume, the related engine will be created in following `ReconcileVolumeState()`.